src/main/java/com/amazon/opendistroforelasticsearch/security/support/ConfigHelper.java

```diff
@@ -70,11 +78,36 @@ public static void uploadFile(Client tc, String filepath, String index, CType cT
                 throw new Exception("   FAIL: Configuration for '" + cType.toLCString()
                         + "' failed for unknown reasons. Pls. consult logfile of elasticsearch");
             }
-        } catch (Exception e) {
-            throw e;
         }
     }
-    
+
+    public static Reader createFileOrStringReader(CType cType, int configVersion, String filepath, boolean populateEmptyIfFileMissing) throws Exception {
```

Comment:
Do we need this populateEmptyIfFileMissing logic? Can we add the file and assume the file to exists in the builds like we do for other configs in securityconfig dir ?

Reply:
The intention is to keep it backward compatible so that if customers perform minor version upgrade, they are not impacted by the document not being present. That said, there are following paths where in I can eliminate missing file handling in couple of places.

1. ConfigurationRepository.uploadFile() - This path is invoked only during index bootstrap which means folks performing minor version upgrade would not hit this path as the index already exists. Missing file handling can be removed from here.
2. OpenDistroSecurityAdmin.migrate(),upload() - These paths provide a way to populate missing documents and thus having missing file handling will be useful.

Now I can remove path 1, but it could help remote cases of migration without using SecurityAdmin tool and ensure the index is always populated. Let me know if you strongly feel the need to remove and I can remove it.